### PR TITLE
add function to make env vars case-insensitive

### LINKED ISSUE #12676: nushell is not compatible with lowercase path on windows
### Describe the bug

Some software installations seem to change Path into lowercase path, causing this BUG

### How to reproduce

1. Go to RegEditor
2. `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Session Manager\Environment`
3. Rename `Path` as `path`
4. Go to nushell, run `which which`

### Expected behavior

Make nushell compatible with lowercase path.

### Screenshots

_No response_

### Configuration

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.92.2                                   |
| branch             |                                          |
| commit_hash        | 2a08a18b26865a86c793d183b9b042220ecb733a |
| build_os           | windows-x86_64                           |
| build_target       | x86_64-pc-windows-msvc                   |
| rust_version       | rustc 1.77.2 (25ef9e3d8 2024-04-09)      |
| rust_channel       | 1.77.2-x86_64-pc-windows-msvc            |
| cargo_version      | cargo 1.77.2 (e52e36006 2024-03-26)      |
| build_time         | 2024-04-10 21:24:37 +00:00               |
| build_rust_channel | release                                  |
| allocator          | mimalloc                                 |
| features           | default, sqlite, trash, which            |
| installed_plugins  |                                          |

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by fdncred**
# Description

This PR adds a new function that allows one to get an env var case-insensitively. I did this so we can hopefully stop having problems when Windows has HKLM as path and HKCU as Path.

Instead of just changing every function that used the original one, I chose the ones that I thought were specific to getting the path. I didn't want to go all in and make every env get case insensitive, but maybe we should? 🤷🏻‍♂️ 

closes #12676

# User-Facing Changes

# Tests + Formatting

# After Submitting

### CODE REVIEW

File: crates/nu-cli/src/completions/command_completions.rs [hunk 1]
```diff
@@ -38,8 +38,7 @@ impl CommandCompletion {
     ) -> Vec<String> {
         let mut executables = vec![];
 
-        // os agnostic way to get the PATH env var
-        let paths = working_set.permanent_state.get_path_env_var();
+        let paths = working_set.permanent_state.get_env_var_insensitive("path");
```
[the remark this answers is not in the document]
Comment (ysthakur) [reply crates/nu-cli/src/completions/command_completions.rs:41]: Gotcha. I'd be against making all env var gets case-insensitive outside of Windows, but I can live with doing that for just PATH.

**Review by IanManske — COMMENTED** [review]
Would you mind using `str::eq_ignore_ascii_case` instead of `str::to_lowercase`? This avoids creating a bunch of new strings. Alternatively, you should use `str::eq_ignore_case` depending on whether Windows is ASCII case insensitive or Unicode case insensitive.

**Comment by fdncred** [comment]
> Would you mind using `str::eq_ignore_ascii_case` instead of `str::to_lowercase`? This avoids creating a bunch of new strings. Alternatively, you should use `str::eq_ignore_case` depending on whether Windows is ASCII case insensitive or Unicode case insensitive for environment variables.

I'm not sure which of these is better. I chose to_lowercase because, from what I read, it works on ascii and unicode. So, I figured the broader application was better. I can look into eq_ignore_case, but I've never used it.

**Comment by IanManske** [comment]
> I'm not sure which of these is better. I chose to_lowercase because, from what I read, it works on ascii and unicode. So, I figured the broader application was better. I can look into eq_ignore_case, but I've never used it.

Yeah, I'm not sure either. In cmd or powershell, do `Å` and `å` overwrite each other / have the same case?

**Comment by fdncred** [comment]
> Yeah, I'm not sure either. In cmd or powershell, do the env vars Å and å overwrite each other / are they treated case insensitively.

I can check but my PR is really about Path PATH path PaTh and I tried to be particular about only changing the functions that dealt with that vs making all env vars case insensitive.

**Comment by devyn** [comment]
From within Nushell, all reads of env vars are case insensitive now, right? Is there ever any reason to want to check env vars case sensitively from inside the codebase?

**Comment by fdncred** [comment]
> From within Nushell, all reads of env vars are case insensitive now, right? Is there ever any reason to want to check env vars case sensitively from inside the codebase?

Maybe internally but not at startup apparently, because if you're on Windows and your HKLM is Path and your HKCU is path things break, or you get half your paths, weirdness ensues. I think this may be only at startup though, which is why I tried to change only those parts. All other internal reading of env vars may be case insensitive. I know I can do $env.path/PATH/PaTh on windows and it reads fine. Seems like some folks don't want case-insensitive everything env vars but I'm not sure why.

**Comment by fdncred** [comment]
We need to dogfood this a bit, let's land and do that.

**Pull request merged**